### PR TITLE
Math-Magicians: Events

### PR DESCRIPTION
**Pull request by Estherstarr**
- [ ] Install big.js
- [ ] Add calculate.js and operate.js files into a logic directory
- [ ] Modify JSX tags in the Calculator.js file
- [ ] Add events to the calculator buttons of the Calculator.js file

### CODE REVIEW

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hi @Estherstarr    
 
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job so far :clap: 
 
 ## Required Changes 
- [ ] Unfortunately, I cannot run your project, can you please take a step back and check it:

image

- **Please fix it and request another CR**

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Estherstarr** [comment]
It's compiling on my system. Please did you try installing the node modules with npm install
@GzimAsani

**Comment by GzimAsani** [comment]
> It's compiling on my system. Please did you try installing the node modules with npm install
> @GzimAsani

Yes, I did try that, if you think the problem was on my side please ask for another CR

**Comment by Estherstarr** [comment]
Hi, incase you run into issues installing the dependencies, you can delete the package-lock.json file and rerun the install commands  :slightly_smiling_face:

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,43 +1,66 @@
...
+              <button type="button" onClick={() => this.handleClick('5')}>5</button>
+              <button type="button" onClick={() => this.handleClick('6')}>6</button>
             </div>
             <div className="calcNumbers">
-              <button type="button">1</button>
-              <button type="button">2</button>
+              <button type="button" onClick={() => this.handleClick('1')}>1</button>
+              <button type="button" onClick={() => this.handleClick('2')}>2</button>
               <button type="button">3</button>
             </div>
```
Comment (Gambit142) [inline src/components/Calculator.js]: - [ ] You don't have an `onClick` function for **_3_**, so when I press 3, nothing is displayed. Kindly fix it. 👍

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,43 +1,66 @@
...
+        <div className="calcDisplay">{obj.next || obj.operation || obj.total || 0 }</div>
         <div className="calcKeysContainer">
           <div className="calcKeys">
             <div className="calcNumbers">
-              <button type="button">AC</button>
-              <button type="button">+/-</button>
-              <button type="button">%</button>
+              <button type="button" onClick={() => this.handleClick('AC')}>AC</button>
+              <button type="button" onClick={() => this.handleClick('+/-')}>+/-</button>
+              <button type="button" onClick={() => this.handleClick('%')}>%</button>
```
Comment (Gambit142) [inline src/components/Calculator.js]: - [ ] One of the project requirements is that props need to be passed from the parent component (`Calculator`) to a child component. I would advise you to make the buttons you display a component and pass props from the Calculator component to the Button component. You are free to implement this requirement however you deem fit, but Kindly pass props from the Calculator component to the child component. 👍

File: src/components/Calculator.js [hunk 3]
```diff
@@ -1,43 +1,66 @@
...
           </div>
           <div className="calcSymbols">
-            <button type="button" className="calcSymbolsColor">÷</button>
-            <button type="button" className="calcSymbolsColor">×</button>
-            <button type="button" className="calcSymbolsColor">-</button>
-            <button type="button" className="calcSymbolsColor">+</button>
-            <button type="button" className="calcSymbolsColor">=</button>
+            <button type="button" className="calcSymbolsColor" onClick={() => this.handleClick('÷')}>÷</button>
+            <button type="button" className="calcSymbolsColor" onClick={() => this.handleClick('×')}>×</button>
+            <button type="button" className="calcSymbolsColor" onClick={() => this.handleClick('-')}>-</button>
```
Comment (Gambit142) [inline src/components/Calculator.js]: - [ ] When I try to multiply two numbers, I get the error that `'×'` is not an operator. Based on the logic in the **_operate.js_** file the symbol for multiplication used is `'x'` and not `'×'`. Kindly make this change by passing the right string as a parameter for the `this.handleClick` function for multiplication. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Estherstarr 👋, 

Good job so far!

## Highlights
- All linters checks passed 💯 
- Made use of the logic directory provided 💯 
- Use of state 💯 
- Use of lifecycle methods 💯 

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by Estherstarr** [comment]
- [ ] Pass props from the Calculator component to the Button component

**Review by Emmyn5600 — APPROVED** [review]
Hi  @Estherstarr 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! :tada: :clap: :clap: :100:

nodding-yes

## Optional suggestions

- [ ] n/a

##### _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

##### Cheers and Happy coding!:clap::clap::clap:
##### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**